idtype/idtype.go

```diff
@@ -153,12 +164,26 @@ func getDefn(t Type) definition {
 // Any types used *must* be registered. Ideally, call this from your package's
 // init function.
 //
+// The third argument of register are the options. It currently only support one option, setting the plural version
+// of the name. If omitted, the name will have an s appended to generate its plural version.
+//
 // Register panics if it is called twice with the same Type.
-func Register(typ Type, mutable bool, name string) {
+func Register(typ Type, mutable bool, name string, options ...string) {
```

Comment:
Personally for future extensibility I would prefer to do functional options here: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

```go
func Register(typ Type, mutable bool, name string, options ...Options)
```

```go
idtype.Register(typ, false, "category", WithPlural("categories"))
```